### PR TITLE
fix(sdk): improvements & fixes for v2.0.0

### PR DESCRIPTION
**Pull request by iamcrazycoder**
#### What this PR does / why we need it:

This PR includes fixes and improvements for v2.0.0.

### CODE REVIEW

File: packages/sdk/src/instant-trade/InstantTradeBuilder.ts [hunk 1]
```diff
@@ -45,24 +47,33 @@ export default class InstantTradeBuilder extends PSBTBuilder {
...
     }
 
-    const utxo = unspendableUTXOs.find((utxo) =>
-      utxo.inscriptions?.find((i) => i.outpoint === this.inscriptionOutpoint)
-    )
+    const utxo = await this.datasource.getInscriptionUTXO(this.inscription.outpoint)
     if (!utxo) {
-      throw new Error("Inscription not found")
+      throw new Error(`Unable to find UTXO: ${this.inscription.outpoint}`)
     }
```
Comment (kevzzsk) [inline packages/sdk/src/instant-trade/InstantTradeBuilder.ts:77]: the point of this function is to check if the seller still have the inscription and seems like now it is only checking if the inscription exists?

File: packages/sdk/src/modules/JsonRpcDatasource.ts [hunk 2]
```diff
@@ -37,12 +37,20 @@ export default class JsonRpcDatasource extends BaseDatasource {
...
-      throw new Error("Invalid options provided.")
+  async getInscriptionUTXO(id: string) {
+    if (!id) {
+      throw new Error("Invalid request")
+    }
+
+    return rpc[this.network].call<UTXO>("Ordinals.GetInscriptionUtxo", { id }, rpc.id)
+  }
+
+  async getInscriptions(owner: string, decodeMetadata = false) {
```
Comment (kevzzsk) [inline packages/sdk/src/modules/JsonRpcDatasource.ts:48]: i think we still need `outpoint` here

File: packages/sdk/src/transactions/Inscriber.ts [hunk 3]
```diff
@@ -232,7 +259,7 @@ export class Inscriber extends PSBTBuilder {
     this.restrictUsageInPreviewMode()
     this.isBuilt()
 
-    const amount = this.fee + this.outputAmount
+    const amount = this.recovery ? this.outputAmount - this.fee : this.outputAmount + this.fee
```
Comment (kevzzsk) [inline packages/sdk/src/transactions/Inscriber.ts:262]: do we not need to add `postage` here like previous fn ? or is it already added into one of the amount

Comment (iamcrazycoder) [reply packages/sdk/src/transactions/Inscriber.ts:262]: postage is already added in the `outputs` which is then reduced and summed into `outputAmount`.

File: packages/sdk/src/modules/JsonRpcDatasource.ts [hunk 4]
```diff
@@ -37,12 +43,51 @@ export default class JsonRpcDatasource extends BaseDatasource {
...
+    mimeSubType,
+    outpoint,
+    decodeMetadata,
+    sort = "asc",
+    limit = 25,
+    next = null
+  }: GetInscriptionsOptions) {
+    if (!owner) {
+      throw new Error("Invalid request")
+    }
```
Comment (kevzzsk) [inline packages/sdk/src/modules/JsonRpcDatasource.ts:67]: i think we need to remove this check?

File: packages/sdk/src/modules/JsonRpcDatasource.ts (hunk 2 above)
Comment (iamcrazycoder) [reply packages/sdk/src/modules/JsonRpcDatasource.ts]: restored

File: packages/sdk/src/modules/JsonRpcDatasource.ts (hunk 4 above)
Comment (iamcrazycoder) [reply packages/sdk/src/modules/JsonRpcDatasource.ts:67]: removed

File: packages/sdk/src/instant-trade/InstantTradeBuilder.ts (hunk 1 above)
Comment (iamcrazycoder) [reply packages/sdk/src/instant-trade/InstantTradeBuilder.ts]: added ownership check

Comment (iamcrazycoder) [reply packages/sdk/src/instant-trade/InstantTradeBuilder.ts:77]: removed ownership check. I will add it back in a separate PR.

**Pull request merged**